File: solutions/java/high-scores/1/src/main/java/HighScores.java
import java.util.Collections;
import java.util.Comparator;
import java.util.List;
import java.util.NoSuchElementException;
import java.util.stream.Collectors;

class HighScores {

    List<Integer> highScores;

    public HighScores(List<Integer> highScores) {
        this.highScores = highScores;
    }

    List<Integer> scores() {
        return highScores;
    }

    Integer latest() {
        return highScores.get(highScores.size()-1);
    }

    Integer personalBest() {
        return highScores.stream().max(Integer::compare).orElseThrow(NoSuchElementException::new);
    }

    List<Integer> personalTopThree() {
        return highScores.stream().sorted(Comparator.reverseOrder()).limit(3).collect(Collectors.toList());
    }
}

File: solutions/dart/high-scores/1/lib/high_scores.dart
class HighScores {
  List<int> _scores;

  HighScores(this._scores);

  List<int> get scores => List.unmodifiable(_scores);

  int latest() => _scores.last;

  int personalBest() => _scores.reduce((a, b) => a > b ? a : b);

  List<int> personalTopThree() => ([..._scores]..sort((a, b) => b.compareTo(a))).take(3).toList();
}
